src/components/ContractSourceCode.tsx

```diff
@@ -69,7 +71,7 @@ function ContractSourceCode() {
 
   useEffect(() => {
     setValue(!!contractProof?.hasOnchainProof ? 0 : 1);
-  }, [contractProof]);
+  }, [contractAddress]);
```

Comment:
```suggestion
  }, [contractProof?.hasOnchainProof]);
```

Using `contractAddress` as a dependency works, but it's because App.tsx happens to have a condition to render this component after proof data has been loaded. it's not safe enough to rely on this.

Using `contractProof.hasOnchainData` is better because before the object kept changing, but here it's just a boolean so it doesn't change as often and triggers the effect